tensorflow_lite_support/ios/test/task/audio/audio_classifier/TFLAudioClassifierTests.m

```diff
@@ -0,0 +1,226 @@
+
+  return [AVAudioPCMBuffer loadPCMBufferFromFileWithPath:filePath
+                                        processingFormat:processingFormat];
+}
+
+- (void)testInferenceWithFloatBufferSucceeds {
+  TFLAudioClassifierOptions *options = [[TFLAudioClassifierOptions alloc] initWithModelPath:self.modelPath];
+  TFLAudioClassifier *audioClassifier = [TFLAudioClassifier audioClassifierWithOptions:options
+                                                                                 error:nil];
+  XCTAssertNotNil(audioClassifier);
+
+  // Create the audio tensor using audio classifier.
+  TFLAudioTensor *audioTensor = [audioClassifier createInputAudioTensor];
+  XCTAssertNotNil(audioTensor);
+
+  // Load pcm buffer from file.
+  AVAudioPCMBuffer *buffer = [self bufferFromFileWithName:@"speech"
+                                                extension:@"wav"
+                                              audioFormat:audioTensor.audioFormat];
+
+  // Get float buffer from pcm buffer.
+  TFLFloatBuffer *floatBuffer = buffer.floatBuffer;
+  XCTAssertNotNil(floatBuffer);
+
+  // Load float buffer into the audio tensor.
+  [audioTensor loadBuffer:floatBuffer offset:0 size:floatBuffer.size error:nil];
+
+  // Perform classification on audio tensor.
+  TFLClassificationResult *classificationResult =
+      [audioClassifier classifyWithAudioTensor:audioTensor error:nil];
+
+  const NSInteger expectedClassificationsCount = 1;
+  VerifyClassificationResult(classificationResult, expectedClassificationsCount);
+
+  const NSInteger expectedHeadIndex = 0;
+  const NSInteger expectedCategoryCount = 521;
+  VerifyClassifications(classificationResult.classifications[0], expectedHeadIndex,
+                        expectedCategoryCount);
+  VerifyCategory(classificationResult.classifications[0].categories[0],
+                 0,          // expectedIndex
+                 0.957031,   // expectedScore
+                 @"Speech",  // expectedLabel
+                 nil         // expectedDisplaName
+  );
+  VerifyCategory(classificationResult.classifications[0].categories[1],
+                 500,                    // expectedIndex
+                 0.019531,               // expectedScore
+                 @"Inside, small room",  // expectedLabel
+                 nil                     // expectedDisplaName
+  );
+  // The 3rd result is different from python tests because of the audio file format conversions are
+  // done using iOS native classes to mimic audio record behaviour. The iOS native classes handle
+  // audio format conversion differently as opposed to the task library C++ convenience method.
+  VerifyCategory(classificationResult.classifications[0].categories[2],
+                 485,          // expectedIndex
+                 0.003906,     // expectedScore
+                 @"Clicking",  // expectedLabel
+                 nil           // expectedDisplaName
+  );
+}
+
+- (void)testInferenceWithAudioRecordSucceeds {
```

Comment:
It is possible for some functionality. I have added a helper for one possible abstraction [here](https://github.com/khanhlvg/tflite-support/blob/ios-audio-classifier/tensorflow_lite_support/ios/test/task/audio/audio_classifier/TFLAudioClassifierTests.m#L185). Using helper methods for creating the classifier, audio tensor etc which are used by all methods will result in needing the created objects to be returned by the helpers. This may impact readability IMO since these are unit tests.

Reply:
Looks great! Thank you